datalad/distribution/siblings.py

```diff
@@ -0,0 +1,382 @@
+            doc="""mode""",
+            constraints=EnsureChoice('query', 'add', 'remove', 'configure') | EnsureNone()),
+        url=Parameter(
+            args=('--url',),
+            doc="""the URL of or path to the dataset sibling named by
+                `name`. For recursive operation it is required that
+                a template string for building subdataset sibling URLs
+                is given.\n List of currently available placeholders:\n
+                %%NAME\tthe name of the dataset, where slashes are replaced by
+                dashes.""",
+            constraints=EnsureStr() | EnsureNone(),
+            nargs="?"),
+        pushurl=Parameter(
+            args=('--pushurl',),
+            doc="""in case the `url` cannot be used to publish to the dataset
+                sibling, this option specifies a URL to be used instead.\nIf no
+                `url` is given, `pushurl` serves as `url` as well.""",
+            constraints=EnsureStr() | EnsureNone()),
+ 
+
+        ## info options
+        # --description gh-1484
+        # --template/cfgfrom gh-1462 (maybe also for a one-time inherit)
+        # --wanted gh-925 (also see below for add_sibling approach)
+
+        fetch=Parameter(
+            args=("--fetch",),
+            action="store_true",
+            doc="""fetch the sibling after configuration"""),
+        as_common_datasrc=as_common_datasrc,
+        publish_depends=publish_depends,
+        publish_by_default=publish_by_default,
+        annex_wanted=annex_wanted_opt,
+        annex_group=annex_group_opt,
+        annex_groupwanted=annex_groupwanted_opt,
+        inherit=inherit_opt,
+
+        recursive=recursion_flag,
+        recursion_limit=recursion_limit)
+
+    @staticmethod
+    @datasetmethod(name='siblings')
+    @eval_results
+    def __call__(
+            mode='query',
+            dataset=None,
+            name=None,
+            url=None,
+            pushurl=None,
+            # TODO consider true, for now like add_sibling
+            fetch=False,
+            as_common_datasrc=None,
+            publish_depends=None,
+            publish_by_default=None,
+            annex_wanted=None,
+            annex_group=None,
+            annex_groupwanted=None,
+            inherit=False,
+            recursive=False,
+            recursion_limit=None):
+
+        # TODO catch invalid mode specified
+        mode_worker_map = {
+            'query': _query_remotes,
+            'add': _add_remote,
+            'configure': _configure_remote,
+            'remove': _remove_remote,
+        }
+        # all worker strictly operate on a single dataset
+        # anything that deals with hierarchies and/or dataset
+        # relationships in general should be dealt with in here
+        # at the top-level and vice versa
+        worker = mode_worker_map[mode]
+
+        dataset = require_dataset(
+            dataset, check_installed=False, purpose='sibling configuration')
```

Comment:
I think it is fine. Any operation is same flavor of configuration, even removal.